modules/ethereum/src/lib.rs

```diff
@@ -83,6 +99,33 @@ pub struct StoredHeader<Submitter> {
 	/// this is the set that has produced the block itself.
 	/// The hash is the hash of block where validators set has been enacted.
 	pub next_validators_set_id: u64,
+	/// Hash of the last block which has **SCHEDULED** validators set change.
+	/// Note that signal doesn't mean that the set has been (or ever will be) enacted.
+	/// Note that the header may already be pruned.
```

Comment:
What are the implications if the header has already been pruned?

Reply:
You can't read header by the hash :)